opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -1959,10 +1962,181 @@ std::vector<double> EclipseGrid::createDVector(const std::array<int,3>& dims, st
     }
 
     const std::vector<int>& EclipseGrid::getActiveMap() const {
-
         return m_active_to_global;
     }
 
+    void EclipseGrid::assertLabelLGR(const std::string& label) const{
+        if (std::find(all_lgr_labels.begin(), all_lgr_labels.end(), label) == all_lgr_labels.end()){
+            throw std::invalid_argument("LGR label not found");
+        }            
+    }
+
+
+    void EclipseGrid::assertIndexLGR(size_t localIndex) const {
+     if (std::find(lgr_active_index.begin(), lgr_active_index.end(), localIndex) != lgr_active_index.end()) {
+        throw std::invalid_argument("input provided is an LGR refined cell");
+      } 
+    }
+
+    std::size_t EclipseGrid::getActiveIndexLGR(std::string label, std::size_t localIndex) const {
+        assertLabelLGR(label);
+        return ActiveIndexLGR(label, localIndex);
+    };
+
+    size_t EclipseGrid::getActiveIndexLGR(std::string label, size_t i, size_t j, size_t k) const {
+        assertLabelLGR(label);
+        return ActiveIndexLGR(label,i,j,k);
+    }
+
+    size_t EclipseGrid::ActiveIndexLGR(std::string label, std::size_t localIndex) const {
+        if (lgr_label.compare(label) == 0 ){
```

Comment:
I'm guessing that this is the case for cells on the GLOBAL grid that have not been involved in any refinement, so they do not have children, and the lgr_label then is the defualt one? GLOBAL. Is this somehow the idea?